### PR TITLE
Fiber dev mode detection using a test function

### PR DESCRIPTION
**Pull request by prometheansacrifice**
Attempting fix for #694. Yet to test creating bundle using [patched react](https://github.com/facebook/react/pull/9661). Will post the results here.

### CODE REVIEW

File: backend/installGlobalHook.js [hunk 1]
```diff
@@ -42,15 +42,27 @@ function installGlobalHook(window: Object) {
...
 
-        // FIXME: this is fragile!
-        // We should replace this check with check on a specially passed
-        // function. This also doesn't detect lack of dead code elimination
-        // (although this is not a problem since flat bundles).
-        if (findFiberCode.indexOf('getClosestInstanceFromNode') !== -1) {
+
+        // DEV mode check
+        if (testMinification() === 42) {
+          // __DEV__ block not removed. Do we warn in console?
```
Comment (prometheansacrifice) [inline backend/installGlobalHook.js]: @gaearon Need a little help here. If `__DEV__` is not removed do we warn?

**Comment by prometheansacrifice** [comment]
Failing flow check. Is it okay to change the type of `renderer` to `any`?

**Comment by gaearon** [comment]
No, please add the new method to renderer definition in `types.js.`

**Comment by prometheansacrifice** [comment]
Not that it is necessary here, but what about [`getInspectorDataForViewTag`](https://github.com/facebook/react/blob/master/src/renderers/native/ReactNativeFiber.js#L123) and in general additions we make to `renderer` in future?

**Pull request closed without merging**